lib/src/arithmetic/fixed_to_float.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_to_float.dart
+// Transform fixed point signals to floating point signals.
+//
+// 2024 October 24
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// [FixedToFloat] converts a fixed point input to floating point with
+/// rouding to nearest even. If the input exceeds the floating point range,
+/// the output will be infinity.
+class FixedToFloat extends Module {
+  /// Width of exponent, must be greater than 0.
+  final int exponentWidth;
+
+  /// Width of mantissa, must be greater than 0.
+  final int mantissaWidth;
+
+  /// Internal representation of the output port.
+  late final FloatingPoint _float =
+      FloatingPoint(exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+  /// Output port [float]
+  late final FloatingPoint float = _float.clone()..gets(output('float'));
+
+  /// Constructor
+  FixedToFloat(FixedPoint fixed,
+      {required this.exponentWidth,
+      required this.mantissaWidth,
+      super.name = 'FixedToFloat'}) {
+    fixed = fixed.clone()..gets(addInput('fixed', fixed, width: fixed.width));
+    addOutput('float', width: _float.width) <= _float;
+
+    final bias = FloatingPointValue.computeBias(exponentWidth);
+    final eMax = pow(2, exponentWidth) - 2;
+    final iWidth =
+        (2 + max(fixed.n, max(log2Ceil(fixed.width), exponentWidth))).toInt();
+
+    // Special handling needed for E4M3 as it does not support inf
+    if ((exponentWidth == 4) && (mantissaWidth == 3)) {
+      UnimplementedError('E4M3 is not supported.');
```

Comment:
It's probably a good idea to explicitly note this in the constructor doc string rather than let people discover it after using it